### ISSUE TITLE
`Fatal RPython error: AssertionError` regression on 3.10 when trailing backslash

### ISSUE THREAD

**Issue by gpotter2** [pypy/pypy#5076]
There's a regression on 3.10 in the parser with trailing backslashes. This is almost literaly the same bug as https://github.com/pypy/pypy/issues/4002 with the only difference being that it's now in a string :P

**Python 3.10.14 [PyPy 7.3.18-alpha0 with GCC 10.2.1 20210130 (Red Hat 10.2.1-11)**: (yesterday's nightly)
```python
>>> import code
>>> code.InteractiveInterpreter().runsource("a = '\\")
RPython traceback:
  File "pypy_interpreter_pyparser.c", line 3407, in PegParser__parse
  File "pypy_interpreter_pyparser.c", line 7725, in generate_tokens
Fatal RPython error: AssertionError
Aborted (core dumped)
```

**Python 3.9.19 [PyPy 7.3.16 with GCC 10.2.1 20210130 (Red Hat 10.2.1-11)] on linux**:
```python
>>>> import code
>>>> code.InteractiveInterpreter().runsource("a = '\\")
True
```

**Python 3.11.2 [CPython]**:
```python
>>> import code
>>> code.InteractiveInterpreter().runsource("a = '\\")
True
```

Feel free to ask for more info. Cheers

**Comment by mattip** [comment]
Here is the last couple of lines from the exception when running the reproducer under `python2 pypy/bin/pyinteractive`:
```
  File "<2512-codegen /home/matti/oss/pypy/rpython/tool/sourcetools.py:174>", line 3, in _run
  File "/home/matti/oss/pypy/pypy/module/__builtin__/compiling.py", line 86, in compile
    optimize=optimize)
  File "/home/matti/oss/pypy/pypy/interpreter/pycompiler.py", line 180, in compile
    mod = self._compile_to_ast(source, info)
  File "/home/matti/oss/pypy/pypy/interpreter/pycompiler.py", line 163, in _compile_to_ast
    mod = self.parser.parse_source(source, info)
  File "/home/matti/oss/pypy/pypy/interpreter/pyparser/pyparse.py", line 180, in parse_source
    return self._parse(textsrc, compile_info)
  File "/home/matti/oss/pypy/pypy/interpreter/pyparser/pyparse.py", line 198, in _parse
    tokens = pytokenizer.generate_tokens(source_lines, flags)
  File "/home/matti/oss/pypy/pypy/interpreter/pyparser/pytokenizer.py", line 206, in generate_tokens
    assert strstart[3] # must be triple-quoted
AssertionError
```

**Comment by mattip** [comment]
Fixed in #5708

**Comment by gpotter2** [comment]
Wow, that was fast ! Thanks a lot
